File: data/code_pairs/fortran/1104.f90
program test
  implicit none
  integer :: n, result

  n = 100
  result = parallel_sum(n)

  if (result /= n) then
    print *, "Error: The sum did not match the expected value."
    call abort
  else
    print *, "Test Passed: Sum is ", result
  endif
contains
  function parallel_sum(n) result(var)
    integer, intent(in) :: n
    integer :: var, i

    var = 0
    !$acc parallel loop reduction(+:var)
    do i = 1, n
       var = var + 1
    end do
    !$acc end parallel loop
  end function parallel_sum
end program test

File: data/code_pairs/cpp/1104.cpp
#include <iostream>
#include <omp.h>

int parallel_sum(int n) {
    int var = 0;

    #pragma omp parallel for reduction(+:var)
    for(int i = 1; i <= n; ++i) {
        var += 1;
    }

    return var;
}

int main() {
    int n = 100;
    int result = parallel_sum(n);

    if(result != n) {
        std::cerr << "Error: The sum did not match the expected value." << std::endl;
        abort();
    }

    std::cout << "Test Passed: Sum is " << result << std::endl;
    return 0;
}